BLSQ/iaso
# Polio 1151 lqas afro map round data

LQAS Afro Map was not filtering future rounds properly

Related JIRA tickets : POLIO-1151

## Self proofreading checklist


## Changes

- Add method get_latest_active_campaign_and_rounds` to `LqasAfroViewset` to filter out:
    - future rounds
    - rounds with `lqas_ended_at` in the future (i.e: LQAs not over). If `lqas_ended_at` is null, we use `ended_at` + 10 days (same buffer as in the openhexa pipeline)

## How to test

- run iaso locally and use ngrok to allow openhexa to access your localhost
- clone https://github.com/BLSQ/iaso-pipelines/. checkout to version 63 (via git tags)
- Run the pipeline with your localhost as target
- In iaso, go to campaigns
- In an existing campaign, add a round with an end date in the future
- Go to afro map and look for the country of the campaign you just updated: the future round should not be displayed
- repeat with: 
    -  an lqas end date in the future (should not appear on the map)
    - no lqas end date and an end date more than 10 days in the past (should appear)
    - no lqas end date and and end date less than 10 days in the past (should not appear)
    - an lqas end date and a round end date less than 10 days in the past (should appear) 

## Print screen / video


https://github.com/BLSQ/iaso/assets/38907762/b322d9c2-52da-4e95-8ede-a1985e41eb9c
